spark/v3.3/spark-extensions/src/test/java/org/apache/iceberg/spark/extensions/TestAddFilesProcedure.java

```diff
@@ -394,6 +397,29 @@ public void addPartitionToPartitioned() {
         sql("SELECT id, name, dept, subdept FROM %s ORDER BY id", tableName));
   }
 
+  @Test
```

Comment:
We should add a test where addfiles is called twice

Reply:
I tried that and it was kinda hard to reproduce as we need spark to generate the same stageId and taskAttemptId.

I can change it into a integration test and try to reproduce it.

Reply:
Can you try doing it like Kafka does it?
https://github.com/apache/spark/blob/cfd7ca9a06161f7622b5179a777f965c11892afa/external/kafka-0-10/src/test/scala/org/apache/spark/streaming/kafka010/KafkaDataConsumerSuite.scala#L96-L99
No need to reset, just inject another task context?

Reply:
I think you just make a test method which doesn't call TaskContext.get and instead passes in a TaskContext

Reply:
Or you can just see if you can make a new thread and set a new context? I think there are some ways to test this without a full integration test.

Reply:
Also even if we don't get a repo with two "add_files" let's add that as a test

Reply:
got it, let me try that. Thanks for the pointers

Reply:
@RussellSpitzer  updated the test to do two add_file calls and verify both partition data is available. kept the check for uuid pattern in manifest path